examples/datetime-timezone/requirements.txt

```
@@ -0,0 +1 @@
+Flask-Admin
```

The example also needs sqlalchemy and flask-sqlalchemy-lite installed, so can we change this please:

```suggestion
.[sqlalchemy]
flask-sqlalchemy-lite
```

It also raises a slightly interesting question. Installing `flask-admin[sqlalchemy]` will install `flask-sqlalchemy` as a dependency - but this uses `flask-sqlalchemy-lite` instead. From a quick look, I think us installing `flask-sqlalchemy` automatically is not actually required. If that's true, we should probably remove it from our dependencies. I'll need to check a bit closer though.

---

Thanks! Flask-Admin does not install flask-sqlalchemy as a dependency. Just tested with conda:
`conda create -n flask-admin-venv`
`conda install flask-admin`
`conda list`
gives
```

_libgcc_mutex
_openmp_mutex
blinker
bzip2
ca-certificates
click
flask
flask-admin
importlib-metadata
itsdangerous
jinja2
ld_impl_linux-64
libexpat
libffi
libgcc-ng
libgomp
libnsl
libsqlite
libuuid
libxcrypt
libzlib
markupsafe
ncurses
openssl
pip
python
python_abi
readline
setuptools
tk
tzdata
werkzeug
wheel
wtforms
xz
zipp
```

---

I have added flask-sqlalchemy-lite as a dependency who has sqlalchemy as a required dependency. https://github.com/pallets-eco/flask-admin/pull/2502/commits/73cff4366a3434897634553e1a3c975b9ec4e81c

---

Ah - currently it doesn't (v1.6.1), but as of v2.0.0 it's listed as a dependency, so would be installed automatically.

---

> Ah - currently it doesn't (v1.6.1), but as of v2.0.0 it's listed as a dependency, so would be installed automatically.

Ah! I missed that. flask-sqlalchemy is used only in /flask_admin/tests, so I agree it should not be a dependency. https://github.com/pallets-eco/flask-sqlalchemy-lite is a very new project from july but is part of pallets-eco. Obviously I can use an except ImportError to use the more established flask-sqlalchemy instead

---

My mistake - didn't spot that you'd added a separate TimezoneAware Article tab now.

---

Yeah, I've seen it mentioned a lot recently. We should probably investigate adding direct support for it (if we need to do anything).

I think that anyone using Flask-admin with SQLAlchemy will have been using Flask-SQLAlchemy. For now, could we keep the example consistent with the rest and just use Flask-SQLAlchemy instead of the new `-Lite` project?

Something for us/me to look into in the future around adding proper support/removing the Flask-SQLAlchemy dependency.

---

done in https://github.com/pallets-eco/flask-admin/pull/2502/commits/cf679e04684b6268e1d0fb863db30d4f4ec3ebcc .

>Yeah, I've seen it mentioned a lot recently. We should probably investigate adding direct support for it (if we need to do anything). [...] Something for us/me to look into in the future around adding proper support/removing the Flask-SQLAlchemy dependency.

I expect close to no work in Flask-Admin to support flask-sqlalchemy-lite as flask-sqlalchemy is imported only in 3 tests and pagination is handled using native sqlalchemy rather than flask-sqlalchemy one. I am trying out flask-sqlalchemy-lite and works fine with flask-admin.
For now, I would only advocate for dropping flask-sqlalchemy as a required flask-admin dependency since it's not required (e.g. pewee does not require flask-sqlalchemy and is supported by flask-admin).